### PR TITLE
change inactive contracts query

### PR DESCRIPTION
**Pull request by MehdiElrafi**

### CODE REVIEW

**Comment by MehdiElrafi** [comment]
[When i try to edit an employee information, an error is displayed](https://trello.com/c/jooG0L1t/196-when-i-try-to-edit-an-employee-information-an-error-is-displayed)

File: app/views/users/_form.html.erb [hunk 1]
```diff
@@ -135,9 +135,9 @@
           <div class="tab-content" id="contract-tab">
             <div data-controller="nested-form" id="contract">
               <div class="border-b border-gray-900/10 pb-10 grid grid-cols-1 gap-x-6 gap-y-8 sm:grid-cols-6">
-                <% if (action_name == 'edit' || action_name == 'update')  && @contracts.any? == true %>
-                  <% inactive_contract = @contracts.order(:created_at).limit(@contracts.count - 1) %>
-                  <% inactive_contract.each do |inactive_contract| %>
+                <% if (action_name == 'edit' || action_name == 'update') %>
+                  <% inactive_contracts = @contracts.where.not(id: @contracts.last).order(:created_at) %>
```
Comment (mohamedrez) [inline app/views/users/_form.html.erb:139]: I don't think this should  be handled in the view, what do you mean by inactive contract

**Review by mohamedrez — CHANGES_REQUESTED** [review]

File: app/views/users/_form.html.erb (hunk 1 above)
Comment (MehdiElrafi) [reply app/views/users/_form.html.erb:139]: In the Contract model I store the history of user's contracts, I display this history above contract form (Inactive contracts means all contracts excluding last one)

Comment (mohamedrez) [reply app/views/users/_form.html.erb:139]: you need a field status in the DB for the contract that sets it to innactive automantically, and sets the new one to actif

File: db/schema.rb [hunk 2]
```diff
@@ -211,11 +212,13 @@
     t.string "title_of_function"
     t.date "date_of_assuming_the_position"
     t.date "date_of_ending_the_position"
-    t.string "service"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.bigint "site_id"
     t.bigint "user_id"
+    t.integer "manager_id"
+    t.bigint "service_id", null: false
```
Comment (mohamedrez) [inline db/schema.rb:220]: is this migration supposed to be here ?

**Review by mohamedrez — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 2 above)
Comment (MehdiElrafi) [reply db/schema.rb:220]: It's my bad ! I will resolve that

**Review by mohamedrez — APPROVED** [review]

**Pull request merged**